### PR TITLE
Improve array, typed array indexing performance

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/ArrayObject.h [hunk 1]
```diff
@@ -128,9 +131,17 @@ class ArrayObject : public Object {
         return m_arrayLength;
     }
 
+    bool hasFastModeDataBuffer()
+    {
+#if defined(ESCARGOT_64) && defined(ESCARGOT_USE_32BIT_IN_64BIT)
```
Comment (clover2123) [inline src/runtime/ArrayObject.h:136]: Is it possible to replace existing `isFastArrayMode` method by this new `hasFastModeDataBuffer`?

Comment (ksh8281) [reply src/runtime/ArrayObject.h:136]: if array length is 0 and array is length
there is no fast mode data buffer..

Comment (clover2123) [reply src/runtime/ArrayObject.h:136]: I get it

**Review by clover2123 — APPROVED** [review]

File: src/runtime/ArrayBuffer.h [hunk 2]
```diff
@@ -116,7 +119,24 @@ class ArrayBuffer : public Object {
         GC_set_bit(desc, GC_WORD_OFFSET(ArrayBuffer, m_backingStore));
     }
 
+    void updateBackingStore(Optional<BackingStore*> bs)
+    {
+        m_backingStore = bs;
+        m_cachedDataAddress = nullptr;
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/ArrayBuffer.h:125]: Because I want to get address lately if possible
getting address can cause malloc from outside

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**